payments/management/commands/expire_too_old_unpaid_orders.py

```diff
@@ -1,18 +1,40 @@
 import logging
+import sys
 
+from django.conf import settings
 from django.core.management.base import BaseCommand
 from django.db.transaction import atomic
 
 from payments.models import Order
 
 logger = logging.getLogger(__name__)
 
+# Output results also to stdout so that the kubernetes cronjob results are available
+logger.addHandler(logging.StreamHandler(sys.stdout))
+
 
 class Command(BaseCommand):
     help = 'Sets too old orders from state "waiting" to state "expired".'
 
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--dry-run",
+            action="store_true",
+            help="Dry run, roll back the transaction at end",
+        )
+
     @atomic
-    def handle(self, *args, **options):
+    def handle(self, dry_run, *args, **options):
         logger.info("Expiring too old unpaid orders...")
-        num_of_updated_orders = Order.objects.update_expired()
-        logger.info("Done, {} order(s) got expired.".format(num_of_updated_orders))
+        num_of_updated_orders = Order.objects.expire_too_old_unpaid_orders(
+            settings.EXPIRE_WAITING_ORDERS_OLDER_THAN_DAYS
+        )
+        if dry_run:
+            logger.info(
+                "Dry run, do not save changes {} order(s) would be expired.".format(
+                    num_of_updated_orders
+                )
+            )
+            raise Exception("Initiate rollback")
```

Comment:
Is this the best approach? Would it be safer to just get the "expirable" orders rather than change-and-rollabck?

Reply:
The dry-run option is useful for manual testing in staging (don't need to rebuild test data anew each time). This would be unsafe if the code had side effects that aren't limited to database state (such as, in future, if an email was sent on state change).
But I'll refactor this to not use a rollback

Reply:
done